### PR TITLE
bf16 support for fake_quantize_learnable_per_channel_affine

### PR DESCRIPTION
**Pull request by liangel-02**
Adding bf16 support for `torch._fake_quantize_learnable_per_channel_affine()` op by relaxing the type check on scale

TODO: need to add bf16 support to `per_tensor_affine_` as `torch._fake_quantize_learnable_per_tensor_affine_backward` gets called in the backward pass 

**Test**
Modified unit test in `test_workflow_ops.py`

### CODE REVIEW

File: aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ std::tuple<Tensor, Tensor> fake_quantize_per_channel_affine_cachemask(
     int64_t axis,
     int64_t quant_min,
     int64_t quant_max) {
-  TORCH_CHECK(scale.scalar_type() == ScalarType::Float,
+  TORCH_CHECK(scale.scalar_type() == ScalarType::Float || scale.scalar_type() == at::kBFloat16,
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp:51]: do we need to allow bfloat16 for zero_point as well

File: test/quantization/core/test_workflow_ops.py [hunk 2]
```diff
@@ -823,21 +824,22 @@ def test_learnable_forward_per_channel_cpu(self, X):
...
+
+        num_dims = random.randint(1, 5)
+        shape = tuple(random.randint(1, 10) for _ in range(num_dims))
+        axis = random.randint(0, num_dims - 1)
+
+        for dtype in [torch.float32, torch.bfloat16]:
+            X_base = torch.randn(shape, device="cuda").to(dtype)
+            channel_size = X_base.size(axis)
+            scale_base = torch.normal(mean=0, std=1, size=(channel_size,)).clamp(1e-4, 100).to(dtype)
+            zero_point_base = torch.normal(mean=0, std=128, size=(channel_size,)).to(dtype)
```
Comment (jerryzh168) [inline test/quantization/core/test_workflow_ops.py:836]: will this error? since current code doesn't support zero_point being bfloat16

File: test/quantization/core/test_workflow_ops.py [hunk 3]
```diff
@@ -823,21 +824,22 @@ def test_learnable_forward_per_channel_cpu(self, X):
...
-        X, (_, _, axis, _) = X
-        X_base = torch.tensor(X).to('cuda')
-        channel_size = X_base.size(axis)
-        scale_base = torch.normal(mean=0, std=1, size=(channel_size,)).clamp(1e-4, 100)
-        zero_point_base = torch.normal(mean=0, std=128, size=(channel_size,))
-        self._test_learnable_forward_per_channel(
-            X_base, 'cuda', scale_base, zero_point_base, axis)
+
+        num_dims = random.randint(1, 5)
+        shape = tuple(random.randint(1, 10) for _ in range(num_dims))
```
Comment (jerryzh168) [inline test/quantization/core/test_workflow_ops.py]: `torch.randint`?

File: test/quantization/core/test_workflow_ops.py [hunk 4]
```diff
@@ -823,21 +824,22 @@ def test_learnable_forward_per_channel_cpu(self, X):
...
         torch.random.manual_seed(NP_RANDOM_SEED)
-        X, (_, _, axis, _) = X
-        X_base = torch.tensor(X).to('cuda')
-        channel_size = X_base.size(axis)
-        scale_base = torch.normal(mean=0, std=1, size=(channel_size,)).clamp(1e-4, 100)
-        zero_point_base = torch.normal(mean=0, std=128, size=(channel_size,))
-        self._test_learnable_forward_per_channel(
-            X_base, 'cuda', scale_base, zero_point_base, axis)
+
+        num_dims = random.randint(1, 5)
```
Comment (jerryzh168) [inline test/quantization/core/test_workflow_ops.py]: we can have a fix dim probably

File: aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp (hunk 1 above)
Comment (liangel-02) [reply aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp:51]: zero_point is cast to int32 here https://github.com/pytorch/pytorch/blob/724463d5a2fba369cd14e89215b84d1b01435df7/aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp#L149

i can remove this casting in a separate pr

File: test/quantization/core/test_workflow_ops.py (hunk 2 above)
Comment (liangel-02) [reply test/quantization/core/test_workflow_ops.py:836]: https://github.com/pytorch/pytorch/pull/165098#discussion_r2417961651

**Review by jerryzh168 — APPROVED** [review]

File: aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp [hunk 5]
```diff
@@ -48,7 +48,7 @@ std::tuple<Tensor, Tensor> fake_quantize_per_channel_affine_cachemask(
...
               "Scale must be Float, found ", scale.scalar_type());
```
Comment (andrewor14) [inline aten/src/ATen/native/quantized/FakeQuantPerChannelAffine.cpp]: need to update this message

**Review by andrewor14 — APPROVED** [review]
Thanks, please add a TODO somewhere (PR description is fine) for fixing the per_tensor version
